security/azure/sql_publicly_accessible_managed_instance/check_for_publicly_accessible_azure_sql_managed_instance.pt

```diff
@@ -116,12 +116,57 @@ datasource "ds_list_sql_managed_instances" do
   end
 end
 
+datasource "ds_sql_managed_instances" do
+  run_script $js_sql_managed_instances, $ds_list_sql_managed_instances, $param_exclusion_tag_key
+end
+
+##################
+# Script         #
+##################
+
+script "js_sql_managed_instances", type: "javascript" do
+  parameters "ds_list_sql_managed_instances", "param_exclusion_tag_key"
+  result "res"
+  code <<-EOS
+    res = []
+    for (var j = 0; j < ds_list_sql_managed_instances.length; j++) {
+      var instance = ds_list_sql_managed_instances[j];
+      var tags=instance['tags'];
+      var is_tag_matched=false;
+      var param_exclusion_tag_key_lowercase = param_exclusion_tag_key.toLowerCase();
+        if(typeof(tags) != "undefined" && tags!=null){
+          for(var k=0;k<tags.length;k++){
+            var tag=tags[k];
+              if(param_exclusion_tag_key_lowercase.indexOf((tag['tagKey']).toLowerCase()) !== -1){
+               is_tag_matched = true;
+              }
+          }
+        }         
+        if(!is_tag_matched){
+          res.push({
```

Comment:
please sort by subscriptionName and region/location

Reply:
Sorted by subscriptionName and location